javascript/ql/src/Declarations/DeadStoreOfLocal.ql

```diff
@@ -63,7 +80,7 @@ where
   (
     // To avoid confusion about the meaning of "definition" and "declaration" we avoid
     // the term "definition" when the alert location is a variable declaration.
-    if dead instanceof VariableDeclarator
+    if dead instanceof VariableDeclarator and hasDominatingDef(dead, v)
```

Comment:
The fix doesn't quite address the problem described in the comment above.

This branch was introduced because people thought "This definition of v ..." referred to the variable as a whole and not just its initial assignment. With this change I think we'll just swing back to the old problem. I expect the next FP report to claim that now the other message is confusing, because the variable is read somewhere.

I have a two-fold suggestion:
1. As a simpler way to check dominance, check if the `SsaImplicitInit` node exists for this variable. This exists only if it's live at the entry, i.e. there is a use not dominated by an assignment.
2. If the `SsaImplicitInit` exists, change the message somehow. Maybe just `The value assigned to v here is unused`.